include/nlohmann/detail/input/json_sax.hpp

```diff
@@ -75,6 +76,14 @@ struct json_sax
     */
     virtual bool string(string_t& val) = 0;
 
+    /*!
+    @brief a binary string was read
+    @param[in] val  binary value
+    @return whether parsing should proceed
+    @note It is safe to move the passed binary.
+    */
+    virtual bool binary(binary_t& val) = 0;
```

Comment:
As binary formats may know the size of a binary value in advance, we may want to pass it to `binary` just as we do with arrays and objects. What do you think?

Reply:
Perform byte by byte operations on binary formats?  I mean, it's not an awful idea.  But because an array and an object can contain arbitrary JSON objects, but binary data cannot, it's not nearly as useful.

Reply:
What I meant is that, for instance, CBOR usually knows the number of bytes in a type (unless you parse until a `break`). So if `0x58 0x1f` is read, then we know it's byte string with 31 bytes. Instead of "just" putting the 31 bytes into a `std::vector<std::uint8_t>` and call `binary(val)`, we could call `binary(val, 31)` to help the consumer. But this idea is bogus. I thought about how we process arrays with a `start_array()` call. This is not the case here - whoever has access to `val` can call `val.size()` and get all the information.

Sorry for the confusion.